### ISSUE TITLE
forge test --debug can't differentiate between two tests with the same name

### ISSUE THREAD

**Issue by transmissions11** [gakonst/foundry#1083]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (865bc46 2022-03-28T00:08:15.310912+00:00)

### What command(s) is the bug in?

forge test --debug

### Operating System

macOS (M1)

### Describe the bug

`forge test --debug` isn't able to debug a concrete test that has the same name as a fuzz test

here's what the error looks like:

```
λ forge test --debug 'testSafeCastTo96()'
[⠔] Compiling...
No files changed, compilation skipped
Error:
   0: 2 tests matched your criteria, but exactly 1 test must match in order to run the debugger.



                              Use --match-contract and --match-path to further limit the search.

Location:
   cli/src/cmd/forge/test.rs:240

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

to reproduce run `forge test --debug 'testSafeCastTo96()` in solmate

**Comment by onbjerg** [comment]
Note: This is an issue with the test filter `--match-test`. It only checks against the function name, not the signature.

**Comment by onbjerg** [comment]
Fixed in #1096, however, since we're using regex, the UX for this is kind of lame since you need to escape the parenthesis:

```
forge test --debug "testSomeFunc\(uint256\)"
```
